### PR TITLE
kubeadm: Print all control plane images on init error

### PR DESCRIPTION
**Pull request by rosti**
**What this PR does / why we need it**:

kubeadm init dumps a comprehensive error message on failure. This message
contains some of the control plane images, the fetching of which could have
caused the init failure. Unfortunately, this is not the full list of images.
Most notably, pause, kube-proxy and DNS images are missing.

This fixes the issue, by using GetAllImages. Furthermore, it simplifies the
code by deduplicating some checks already made in GetAllImages.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Inspired by kubernetes/kubeadm#1016, but not fixing it.

**Special notes for your reviewer**:

/cc @kubernetes/sig-cluster-lifecycle-pr-reviews
/area kubeadm
/assign @luxas
/assign @timothysc

**Release note**:

```release-note
kubeadm: Print all control plane images on init error
```

### CODE REVIEW

**Comment by neolit123** [comment]
/ok-to-test
@kubernetes/sig-cluster-lifecycle-pr-reviews

**Review by luxas — COMMENTED** [review]
This is kinda ok I guess, but the DNS and proxy images aren't actually used in this phase if kubeadm fails there...

File: cmd/kubeadm/app/cmd/init.go [hunk 1]
```diff
@@ -397,17 +396,12 @@ func (i *Init) Run(out io.Writer) error {
...
-		if i.cfg.Etcd.Local != nil {
-			ctx["EtcdImage"] = fmt.Sprintf("				- %s", images.GetEtcdImage(i.cfg))
-		} else {
-			ctx["EtcdImage"] = ""
+		ctx := struct {
+			Error  string
+			Images []string
+		}{
+			Error:  fmt.Sprintf("%v", err),
+			Images: images.GetAllImages(i.cfg),
```
Comment (timothysc) [inline cmd/kubeadm/app/cmd/init.go]: If this includes images other than the control plan I think this will be more confusing.

**Comment by rosti** [comment]
@luxas @timothysc you are both right. `GetAllImages` will return kube-proxy and DNS images too and init failure would not be caused by them. What is missing in the current code base is pause image though (I've seen init failures on non-internet connected machines, because of forgotten pause image).

I can extract portions of `GetAllImages` into `GetAllControlPlaneImages` and still contain the full knowledge of images needed for init, separated into `images.go`.

I'll be on a PTO for the rest of the week, so I'll probably finish it off next week.

WDYT?

**Comment by luxas** [comment]
Adding the pause image is totally ok, dns and proxy not IMO.

> I'll be on a PTO for the rest of the week

Take your time and enjoy!

**Comment by kad** [comment]
@luxas in theory, every image that is referenced by manifests maintained inside kubeadm, should be reported. This really helps for air-gap k8s installations.

**Comment by dixudx** [comment]
Roughly lgtm.

**Comment by luxas** [comment]
> @luxas in theory, every image that is referenced by manifests maintained inside kubeadm, should be reported. This really helps for air-gap k8s installations.

@kad we added `kubeadm config images list` exactly for that purpose.

**Comment by rosti** [comment]
This is superseded by #66658 . Closing for that matter.

**Pull request closed without merging**